src/coreclr/jit/llvm.cpp

```diff
@@ -1738,8 +1766,19 @@ bool Llvm::isLlvmFrameLocal(LclVarDsc* varDsc)
 
 void Llvm::storeLocalVar(GenTreeLclVar* lclVar)
 {
-    Type* destLlvmType = getLlvmTypeForVarType(lclVar->TypeGet());
-    Value* localValue = consumeValue(lclVar->gtGetOp1(), destLlvmType);
+    Type* destLlvmType = getLlvmTypeForLclVar(lclVar);
+
+    Value*   localValue;
+
+    // zero initialization check
+    if (lclVar->TypeGet() == TYP_STRUCT && lclVar->gtGetOp1()->IsIntegralConst(0))
+    {
+        localValue = llvm::Constant::getNullValue(getLlvmTypeForStruct(_compiler->lvaGetDesc(lclVar)->GetLayout()));
```

Comment:
```suggestion
        localValue = llvm::Constant::getNullValue(destLlvmType);
```
(Presumably the same thing)